aspnetcore/signalr/security.md

```diff
@@ -5,121 +5,78 @@ description: Learn how to use authentication and authorization in ASP.NET Core S
+* Prevents a client from being able to send a larger message.
+* The server will never need to allocate large buffers to accept messages.
 
-### Buffer management
+If your messages are larger than 32 K, you can increase the limit. Increasing this limit means:
 
-SignalR uses per-connection buffers in order to manage incoming and outgoing messages. By default, SignalR limits these buffers to 32KB. This means the largest possible message a client or server can send is 32KB. This also means the maximum amount of memory consumed by a connection for messages is 32KB. If you know your messages are always smaller than this limit, you can reduce this size to prevent a client from being able to send a larger message and force the server to allocate memory to accept it. Similarly, if you know your messages are larger than this limit, you can increase it. However, be aware that increasing this limit means that the client is able to cause the server to allocate additional memory and may reduce the number of concurrent connections your app can handle.
+* The client can cause the server to allocate large memory buffers.
+* Server allocation of large buffers may reduce the number of concurrent connections.
 
-There are separate limits for incoming and outgoing messages, both can be configured on the [`HttpConnectionDispatcherOptions`](xref:signalr/configuration#configure-server-options) object configured in `MapHub`:
+There are limits for incoming and outgoing messages, both can be configured on the [`HttpConnectionDispatcherOptions`](xref:signalr/configuration#configure-server-options) object configured in `MapHub`:
 
 * `ApplicationMaxBufferSize` represents the maximum number of bytes from the client that the server buffers. If the client attempts to send a message larger than this limit, the connection may be closed.
-* `TransportMaxBufferSize` represents the maximum number of bytes the server can send. If the server attempts to send a message (includes return values from hub methods) larger than this limit, an exception will be thrown.
+* `TransportMaxBufferSize` represents the maximum number of bytes the server can send. If the server attempts to send a message (including return values from hub methods) larger than this limit, an exception will be thrown.
 
-Setting the limit to `0` disables the limit entirely. However, this should be done with extreme caution. Removing the limit allows a client to send a message of any size. This could be used by a malicious client to cause excess memory to be allocated, which could dramatically reduce the number of concurrent connections your app can support.
+Setting the limit to `0` disables the limit. However, disabling the limit exposes the app malicious large message creation. Removing the limit allows a client to send a message of any size. Malicious clients sending large  
```

Comment:
~~However, disabling the limit exposes the app malicious large message creation.~~
Don't think the sentence is useful since we describe it immediately after